website/draft_site/drafts/views.py

```diff
@@ -1,3 +1,185 @@
+            for card in pack:
+                db_card = models.Card(draft=new_draft, name=card.name, phase=phase, start_seat=start_seat)
+                db_card.save()
+
+    # Create human drafter in DB
+    models.Drafter(draft=new_draft, seat=0, bot=False).save()
+    # Create bot drafters in DB
+    # Seat 0 is the human drafter, so start at seat 1
+    for i in range(1, draft_info.num_drafters):
+        # TODO: for now picker state isn't saved to improve performance; we might need to in the future.
+        bot_drafter = Drafter(None, draft_info)
+        db_bot_drafter = models.Drafter(draft=new_draft, seat=i, bot=True, bot_state=pickle.dumps(bot_drafter))
+        db_bot_drafter.save()
+
+    return HttpResponseRedirect(reverse('show_draft', args=[new_draft.id]))
+
+
+# /draft/<int:draft_id>
+def show_draft(request, draft_id):
+    draft = get_object_or_404(models.Draft, pk=draft_id)
+    drafter = models.Drafter.objects.get(draft=draft, seat=0)
+
+    pack_index = _get_pack_index(draft, drafter)
+    cards = models.Card.objects.filter(draft=draft, phase=drafter.current_phase,
+                                       start_seat=pack_index, picked_by__isnull=True)
+    owned_cards = models.Card.objects.filter(draft=draft, picked_by=drafter)
+    sorted_owned_cards = sorted(owned_cards, key=lambda c: (c.phase, c.picked_at))
+
+    context = {'cards': cards, 'draft_id': draft_id,
+               'phase': drafter.current_phase, 'pick': drafter.current_pick,
+               'owned_cards': sorted_owned_cards, 'seats': range(0, draft.num_drafters)}
+
+    return render(request, 'drafts/show_pack.html', context)
+
+
+# TODO: reduce redundancy with show_draft
+# /draft/<int:draft_id>/seat/<int:seat>
+def show_seat(request, draft_id, seat):
+    draft = get_object_or_404(models.Draft, pk=draft_id)
+    drafter = models.Drafter.objects.get(draft=draft, seat=seat)
+
+    pack_index = _get_pack_index(draft, drafter)
+    cards = models.Card.objects.filter(draft=draft, phase=drafter.current_phase,
+                                       start_seat=pack_index, picked_by__isnull=True)
+    owned_cards = models.Card.objects.filter(draft=draft, picked_by=drafter)
+    sorted_owned_cards = sorted(owned_cards, key=lambda c: (c.phase, c.picked_at))
+
+    context = {'cards': cards, 'draft_id': draft_id,
+               'phase': drafter.current_phase, 'pick': drafter.current_pick,
+               'owned_cards': sorted_owned_cards}
+
+    return render(request, 'drafts/show_seat.html', context)
+
+
+# /draft/pick-card/<int:draft_id>
+def pick_card(request, draft_id):
+    try:
+        with transaction.atomic():
+            draft = get_object_or_404(models.Draft, pk=draft_id)
+            drafter = models.Drafter.objects.get(draft=draft, seat=0)
+            picked_card = models.Card.objects.get(id=request.POST['picked_card_id'])
+            phase = int(request.POST['phase'])
+            pick = int(request.POST['pick'])
```

Comment:
The request has no business telling the server what pick or phase it is in; otherwise it could pick from packs which don't exist yet, or change its past pick.

Reply:
Winter showed me that the server handles this correctly on the backend (fails), which seems fine to me.

Reply:
The server still only writes the pick if the current state of the drafter is on that phase and pick; see _save_pick method. I think it makes sense for the client to indicate which pack they'd like to make a pick for, and then the server can say yes or no.